stdlib/list.ml

```diff
@@ -56,6 +56,14 @@ let rec rev_append l1 l2 =
 
 let rev l = rev_append l []
 
+let init n f =
+  if n = 0 then [] else
+  if n < 0 then invalid_arg "List.init" else
+  let rec init_aux acc i =
+    if i >= n then acc else
+    init_aux ((f i)::acc) (i+1)
+  in rev (init_aux [] 0)
```

Comment:
Do we really need the property that `f` is called in the order `0` to `n-1`?
If the answer is no, there is no reason to reverse the list at the end rather than construct the list in reverse order, i.e 
```OCaml
let init_aux f l n = if n = 0 then f 0 :: l else init_aux f (f n :: l) (n-1)
```
If the answer is yes, this property ought to be documented.